### ISSUE TITLE
Pipeline execution hangs when a process exceed the time limit and retry error strategy is used

### ISSUE THREAD

**Issue by huguesfontenelle** [nextflow-io/nextflow#387]
`N E X T F L O W  ~  version 0.25.0.`

Reproducible code for the bug:
```
numbers = Channel.from(1)
process dostuffs {
    echo true
    errorStrategy 'retry'
    maxRetries 1
    time = {'3s' * task.attempt }

    input:
    val(n) from numbers

    script:
"""
echo ${n}
sleep 4
"""
}
```

**local**

_sequential_

The _first_ run of the process exceeds time limit and is rerun (as expected), then the _second_ run of the process hangs forever.

_parallel_

However it I change the first line to numbers = Channel.from([1, 2, 3]), the _first_ run of the processes exceed time limit and are rerun (as expected), then the _second_ run of one of the processes hangs forever.

**slurm**

(edit) untested

Note: similarities with #183

**Comment by pditommaso** [comment]
Is this a regression or it does happen also with a previous NF version ?

**Comment by huguesfontenelle** [comment]
I have corrected my initial description, and here are my observations for nextflow 0.22.0 that I have in production.

`N E X T F L O W  ~  version 0.22.0`

**local**

_sequential_
When submitted to local executor, behaves as expected.

_parallel_
However if I change the first line to `numbers = Channel.from([1, 2, 3])`, then the second run of one of the parallel processes terminates:
````
  Process `dostuffs (3)` terminated for an unknown reason -- Likely it has been terminated by the external system
```

**slurm**

_sequential_
When submitted to local SLURM, behaves as expected.

_parallel_
However it I change the first line to `numbers = Channel.from([1, 2, 3])`, then the second run of one of the parallel processes also times out (exit code 143).
````
  Process `dostuffs (1)` terminated with an error exit status (143)
```

**Comment by huguesfontenelle** [comment]
So yes I observe a change in behaviour between 0.22.0 and 0.25.0, but there's still some work to do :-) 

Whether I submit localy or to slurm, whether I run a sequential process or several parallel ones, I expect: the first processes to fail (they take too much time); the second processes to succeed (they got twice the time with `time = {'3s' * task.attempt }`).

**Comment by huguesfontenelle** [comment]
Just for the context, my case study is this: for FASTQ to BAM mapping of exome, I have assigned 12h. That works well for most samples but occasionally one will need more time. I'd rather not increase the allocated time for all samples (since I'd be billed and lose priority over longer processes somewhat), so I implemented this re-run strategy. It works well for exomes, but here having trouble with trios (in which three mapping run in parallel).

**Comment by pditommaso** [comment]
Thanks for the excellent bug report. I've spotted the problem.

**Comment by pditommaso** [comment]
Uploaded version `0.25.2-SNAPSHOT` that solve the problem.

**Comment by huguesfontenelle** [comment]
Nice!
But are you sure it works for the SLURM parallel case?
That's how I spotted it in first place: my second runs are timing out.

**Comment by pditommaso** [comment]
Yes 

```
N E X T F L O W  ~  version 0.25.2-SNAPSHOT
Launching `t` [adoring_franklin] - revision: 7b05a751bf
[warm up] executor > crg
[6e/a6d752] Submitted process > dostuffs (2)
[6f/4f56cf] Submitted process > dostuffs (1)
[a0/e7e8be] Submitted process > dostuffs (3)
WARN: Process `dostuffs (2)` terminated with an error exit status (140) -- Execution is retried (1)
WARN: Process `dostuffs (1)` terminated with an error exit status (140) -- Execution is retried (1)
[3f/917c9a] Re-submitted process > dostuffs (2)
WARN: Process `dostuffs (3)` terminated with an error exit status (140) -- Execution is retried (1)
[17/12df4e] Re-submitted process > dostuffs (1)
[3d/835765] Re-submitted process > dostuffs (3)
2
1
3
```

**Comment by pditommaso** [comment]
Fixed in version `0.25.2`
